services/src/main/java/org/keycloak/connections/httpclient/ProxyMappings.java

```diff
@@ -106,6 +109,22 @@ public HttpHost getProxyFor(String hostname) {
       .orElse(null);
   }
 
+  /**
+   * @param hostname
+   * @return the proxy credentials associated with the first matching hostname {@link Pattern}
+   * or {@literal null} if none matches.
+   */
+  public String[] getProxyCredentialsFor(String hostname) {
```

Comment:
This would be a heavily used function. Better use some form of a cache (`ConcurrentMap<String, String[]> hostnameToCreds`) instead of dynamic lookup of credentials with every request.

Reply:
I see. Please wait a while.